test/functional/p2p_addr_relay.py

```diff
@@ -282,7 +272,7 @@ def getaddr_tests(self):
         block_relay_peer.send_and_ping(msg_getaddr())
         inbound_peer.send_and_ping(msg_getaddr())
 
-        self.mocktime += 5 * 60
+        self.mocktime += 10 * 60
```

Comment:
This is a bit of a magic number. Perhaps it's worth a comment?

```diff
diff --git a/test/functional/p2p_addr_relay.py b/test/functional/p2p_addr_relay.py
index 5f75670739..1502038b2d 100755
--- a/test/functional/p2p_addr_relay.py
+++ b/test/functional/p2p_addr_relay.py
@@ -272,6 +272,10 @@ class AddrTest(BitcoinTestFramework):
         block_relay_peer.send_and_ping(msg_getaddr())
         inbound_peer.send_and_ping(msg_getaddr())
 
+        # `addr` messages are sent on an exponential distribution with mean interval
+        # of 30s. Setting the mocktime 600s forward and waiting for 60s for the
+        # addr message gives a probability of (1 - e^-(660/30)) that the event will
+        # occur (i.e. this fails less than one in 3 billion repeats).
         self.mocktime += 10 * 60
         self.nodes[0].setmocktime(self.mocktime)
         inbound_peer.wait_until(lambda: inbound_peer.addr_received() is True)
```

Incidentally, I don't like the name `PoissonNextSend` since people often refer to it incorrectly as a "Poisson distribution", whereas it is in fact an _exponential distribution_, which gives the probability distribution of the time between events in a _Poisson point process_ (https://en.wikipedia.org/wiki/Exponential_distribution).

Reply:
I added a comment for that - I think we can't count the last 60s because afaik once we set the mocktime once, the internal time won't progress by waiting.

I think it would make sense to document this feature of PoissonNextSend in `net`, since this has confused me before as well (and others, see e.g. #23347).

Reply:
> I think we can't count the last 60s because afaik once we set the mocktime once, the internal time won't progress by waiting.

Ah yes. I think you're right.

> I think it would make sense to document this feature of PoissonNextSend

I'll have a crack at this.

Reply:
I have a branch here: https://github.com/jnewbery/bitcoin/tree/2021-12-rename-poisson that renames the poisson functions, expands the comments and moves them to more appropriate places. Let me know what you think.